src/GoTrueClient.ts

```diff
@@ -1335,10 +1377,19 @@ export default class GoTrueClient {
     if (options?.scopes) {
       urlParams.push(`scopes=${encodeURIComponent(options.scopes)}`)
     }
+    if (options?.flowType && options.flowType === 'pkce') {
+      urlParams.push(`flow_type=${encodeURIComponent(options.flowType)}`)
+      const codeVerifier = await generatePKCEVerifier()
+      await setItemAsync(this.storage, `${this.storageKey}-oauth-code-verifier`, codeVerifier)
+      const codeChallenge = await generatePKCEChallenge(codeVerifier)
+      urlParams.push(`code_challenge=${encodeURIComponent(codeChallenge)}`)
+      urlParams.push(`code_challenge_method=${encodeURIComponent('s256')}`)
+    }
     if (options?.queryParams) {
       const query = new URLSearchParams(options.queryParams)
       urlParams.push(query.toString())
     }
```

Comment:
Why not just use `new URLSearchParams()` to build the params, instead of these string operations?